test/remoteCollection/collect.js

```diff
@@ -28,687 +28,757 @@ describe('test/remoteCollection/collect.js >', () => {
-      'the "res" attribute', (done) => {
       const generator = {
-        name: 'Generator0',
-        intervalSecs: 1,
+        name: 'generator_ssl',
         context: {},
         generatorTemplate: {
           connection: {
             headers: {
-              Authorization: 'abddr121345bb',
+              Authorization: 'some_data',
             },
-            url: 'notaural',
+            url: 'http://foo.io',
           },
         },
         connection: {},
       };
+
       collect.prepareRemoteRequest(generator)
         .then((collectRes) => {
           expect(collectRes.res).to.not.equal(undefined);
-          expect(collectRes.res.errno).to.equal('ENOTFOUND');
+          expect(collectRes.res.status).to.equal(httpStatus.BAD_REQUEST);
+          const expectedMessage = 'Your Refocus instance is configured to ' +
+            'require SSL for connections to remote data sources. Please ' +
+            'update Sample Generator "generator_ssl" to specify an ' +
+            'https connection url.';
+          expect(collectRes.res.message).to.equal(expectedMessage);
           done();
         })
         .catch(done);
     });
 
-    it('ok, request use data source proxy when present', (done) => {
-      const dataSourceProxy = 'http://abcProxy.com';
-      const remoteUrl = 'http://www.xyz.com/';
-      const generator = {
-        name: 'Generator0',
-        intervalSecs: 1,
-        context: {},
-        generatorTemplate: {
-          connection: {
-            headers: {
-              Authorization: 'abddr121345bb',
-            },
-            url: 'remoteUrl' + '/status',
-            dataSourceProxy,
+  it('retry max times on timeout', (done) => {
+    const remoteUrl1 = 'https://randonUnAvailableUrl.false/';
+    const generator = {
+      name: 'Generator0',
+      intervalSecs: 1,
+      context: {},
+      generatorTemplate: {
+        connection: {
+          headers: {
+            Authorization: 'abddr121345bb',
           },
+          url: 'https://randonUnAvailableUrl.false/',
         },
-        connection: {},
-      };
-
-      nock(remoteUrl)
-        .get('/status')
-        .reply(httpStatus.OK, { status: 'OK' },
-          { 'Content-Type': 'application/json' });
-
-      const spy = sinon.spy(request, 'get');
-      collect.prepareRemoteRequest(generator)
-        .then(() => {
-          expect(spy.returnValues[0]._proxyUri).to.be.equal(dataSourceProxy);
-          spy.restore();
-          done();
-        })
-        .catch((err) => {
-          spy.restore();
-          done(err);
-        });
-    });
+      },
+      connection: {},
+      timeout: {
+        response: 10,
+        deadline: 10,
+      },
+    };
+    const err = new Error('Timeout of 10ms exceeded');
+    err.timeout = 10;
+    err.code = 'ECONNABORTED';
+    nock(remoteUrl1)
+      .get('/')
+      .delay(15)
+      .times(4) // initial req + 3 retries
```

Comment:
So this will mock what should be returned by the request in this test - I am just curious how are we testing that the superagent actually retries on socket hangup when our code executes. Is it manual or am I missing something in the tests?

Reply:
the new test shows that superagent is doing the retry :)